src/components/Popover/index.js

```diff
@@ -0,0 +1,79 @@
+import { Component } from 'preact';
+import { createContext } from 'preact-context';
+import { createClassName, normalizeDOMRect } from '../helpers';
+import styles from './styles';
+
+
+const PopoverContext = createContext();
+
+
+const PopoverOverlay = ({ children, className, visible, ...props }) => (
+	<div
+		className={createClassName(styles, 'popover__overlay', { visible }, [className])}
+		{...props}
+	>
+		{children}
+	</div>
+);
+
+
+export class PopoverContainer extends Component {
+	state = {
+		renderer: null,
+	}
+
+	open = (renderer, props, { currentTarget } = {}) => {
+		let overlayBounds;
+		let triggerBounds;
+
+		if (this.overlayRef) {
+			overlayBounds = normalizeDOMRect(this.overlayRef.getDOMNode().getBoundingClientRect());
```

Comment:
@tassoevan, an error is being raised at this point:

<img width="489" alt="screen shot 2019-01-01 at 16 01 05" src="https://user-images.githubusercontent.com/2067649/50575125-78142e00-0dde-11e9-8a14-4b5ba03d0878.png">